### PR TITLE
{Compute} Fix VM tests

### PR DESCRIPTION
**Pull request by qwordy**
**Description**  

Fix VM tests

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yungezz** [comment]
hi @arrownj  could you pls help to review this? thanks.

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py [hunk 1]
```diff
@@ -75,14 +75,15 @@ def test_vm_images_list_thru_services(self):
 
 class VMOpenPortTest(ScenarioTest):
 
+    @unittest.skip('We don\'t have an environment that allows opening ports. Funny.')
```
Comment (arrownj) [inline src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:78]: :(

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py [hunk 2]
```diff
@@ -2368,6 +2370,7 @@ def test_vmss_existing_lb(self, resource_group):
         self.cmd('network lb create -g {rg} -n {lb} --backend-pool-name test')
         self.cmd('vmss create -g {rg} -n {vmss} --load-balancer {lb} --image UbuntuLTS --admin-username clitester --admin-password TestTest12#$')
 
+    @unittest.skip('Can\'t test lb. Not allowed to open ports')
```
Comment (arrownj) [inline src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:2373]: Why skip it directly ? I'm OK to skip it in live test, however with this decorator, the recording mode will also be skipped.

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:2373]: If can't live, then can't replay.

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py (hunk 1 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:78]: :)

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py (hunk 2 above)
Comment (arrownj) [reply src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:2384]: Can't agree. I think daily live test is different with manual live test. Although this test can not be live run daily, I can still manually run it to generate a recording for it.

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:2384]: We need to disable the policy to run it live and generate a recording for it. It reduces stability of testing results if policies are disabled and enabled frequently.

Comment (arrownj) [reply src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:2384]: So I mean there should be a decorate to skip live test only.

**Review by arrownj — APPROVED** [review]
👍

**Pull request merged**